gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -1696,3 +1791,45 @@ MM_MemoryPoolAddressOrderedList::releaseFreeMemoryPages(MM_EnvironmentBase* env)
 	return releasedBytes;
 }
 #endif
+
+void *
+MM_MemoryPoolAddressOrderedList::alignWithCard(void *address, bool toFloor, uintptr_t alignmentMultiple)
+{
+	uintptr_t alignmentMask = alignmentMultiple - 1;
+	uintptr_t newSum = (uintptr_t)address + alignmentMask;
+	uintptr_t alignedValue = newSum & ~alignmentMask;
+	if (((uintptr_t) address != alignedValue) && toFloor) {
+		alignedValue -= alignmentMultiple;
+	}
+
+	return (void *) alignedValue;
+}
+
+void
+MM_MemoryPoolAddressOrderedList::cleanCardsForFreeEntry(MM_EnvironmentBase *env, void *addrBase, void *addrTop, bool needSync)
```

Comment:
This 2 methods do not belong here, but probably to SweepPoolManagerVLHGC. Pool should not be aware of CT, and possibly aligning as well.
Method should be called 'clear...' not 'clean...' (clearing just sets it to 0, cleaning actually does object scanning to clean the card).
We should try to reuse clearCardsInRange for the middle part.